src/WebJobs.Script/Description/DotNet/CSharp/Analyzers/AsyncVoidAnalyzer.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.Azure.WebJobs.Script.Description.DotNet.CSharp.Analyzers
+{
+    [CLSCompliant(false)]
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class AsyncVoidAnalyzer : DiagnosticAnalyzer
+    {
+        private const string Title = "Avoid Async Void";
+        private const string MessageFormat = "This method has the async keyword but it returns void";
+        private readonly DiagnosticDescriptor _supportedRule;
+
+        public AsyncVoidAnalyzer()
+        {
+            _supportedRule = new DiagnosticDescriptor(DotNetConstants.AsyncVoidCode,
+                Title, MessageFormat, "Function", DiagnosticSeverity.Error, true);
```

Comment:
Saw your comment about this... the diagnostic severity needs to be `Warning`, otherwise we'd fail to compile existing functions that currently compile "just fine"